meta_model_building/get_opt_params.py

```diff
@@ -6,62 +6,200 @@
 from sklearn.model_selection import train_test_split
 import sys
 import time
+import utility
 
-if len(sys.argv) > 1:
-    input_file_name = sys.argv[1]
-    if sys.argv[1] == 'boston':
-        housing = load_boston()
-        data, target = housing.data, housing.target
-    else:
-        try:
-            file_data = np.genfromtxt(input_file_name, delimiter=',')
-            data, target = file_data[:, :-1], file_data[:, -1:] 
-            target = np.ravel(target)
-        except:
-            print("Failed to read data file", input_file_name, "as CSV.  Aborting.")
-            quit()
-else:
+model_list = utility.implemented_model_list
+preprocessor_list = utility.implemented_preprocessor_list
+
+if len(sys.argv) == 1 or sys.argv[1] == 'boston':
     input_file_name = 'boston'
+else:
+    input_file_name = sys.argv[1]
+if input_file_name == 'boston':
     housing = load_boston()
     data, target = housing.data, housing.target
+else:
+    try:
+        file_data = np.genfromtxt(input_file_name, delimiter=',')
+    except:
+        print(
+            "Failed to read data file", input_file_name,
+            "as CSV.  Aborting."
+            )
+        quit()
+    data, target = file_data[:, :-1], file_data[:, -1:]
+    target = np.ravel(target)
+
+if '-seed' in sys.argv:
```

Comment:
`argparse`, used correctly, will REALLY clean up this section.

Reply:
Having now read through the whole bit here... Yeah, look at argparse soon! It'll compactify this to like 10 lines.